app/assets/javascripts/application.js

```diff
@@ -19,185 +19,187 @@
-      $('#temp-3').html(temp3)
-      $('#temp-4').html(temp4)
-      $('#temp-5').html(temp5)
-      $('#temp-6').html(temp6)
-      $('#temp-7').html(temp7)
-      $('#temp-8').html(temp8)
-      $('#day-1').html(day1)
-      $('#day-2').html(day2)
-      $('#day-3').html(day3)
-      $('#day-4').html(day4)
-      $('#day-5').html(day5)
-      $('#icon-1').html(icon1)
-      $('#icon-2').html(icon2)
-      $('#icon-3').html(icon3)
-      $('#icon-4').html(icon4)
-      $('#icon-5').html(icon5)
-      $('#high-1').html("High: " + high1)
-      $('#high-2').html("High: " + high2)
-      $('#high-3').html("High: " + high3)
-      $('#high-4').html("High: " + high4)
-      $('#high-5').html("High: " + high5)
-      $('#day1-icon').attr('src', weatherImages[day_1_Icon])
-      $('#day2-icon').attr('src', weatherImages[day_2_Icon])
-      $('#day3-icon').attr('src', weatherImages[day_3_Icon])
-      $('#day4-icon').attr('src', weatherImages[day_4_Icon])
-      $('#day5-icon').attr('src', weatherImages[day_5_Icon])
+   // Update Dom
+function updateDOM(data) {
+  var weatherImages = {
+    "clear-day": "https://upload.wikimedia.org/wikipedia/commons/thumb/f/fc/Sun_icon.svg/252px-Sun_icon.svg.png",
+    "clear-night": "http://www.clker.com/cliparts/f/S/2/p/7/u/gold-matte-moon.svg",
+    "rain": "https://cdn3.iconfinder.com/data/icons/weather-16/256/Rainy_Day-512.png",
+    "snow": "https://upload.wikimedia.org/wikipedia/commons/thumb/9/93/Nuvola_weather_snow.svg/1000px-Nuvola_weather_snow.svg.png",
+    "sleet": "http://www.clker.com/cliparts/f/6/7/4/1206565674431593790Anonymous_simple_weather_symbols_10.svg.hi.png",
+    "wind": "http://www.haotu.net/up/4233/128/216-wind.png",
+    "fog": "http://www.iconninja.com/files/81/344/943/fog-cloud-hiding-the-sun-weather-interface-symbol-icon.svg",
+    "cloudy": "http://camera.thietbianninh.com/images/icon-2.png",
+    "partly-cloudy-day": "http://meteo.cw/images_www/weather_icons1/weather_icon_03.png",
+    "partly-cloudy-night": "http://icon-park.com/imagefiles/simple_weather_icons_cloudy_night.png",
+    "hail": "http://icons.iconarchive.com/icons/icons8/ios7/256/Weather-Hail-icon.png",
+    "thunderstorm": "http://findicons.com/files/icons/2613/android_weather_extended/480/thunderstorms.png",
+    "tornado": "http://hddfhm.com/images/clipart-of-a-tornado-11.png"
+  }
+  var location = data.location;
+  var currentTemp = data.currently.current_temperature;
+  var todaysHigh = data.currently.high;
+  var todaysLow = data.currently.low;
+  var currentTime = data.currently.current_time;
+  var currentIcon = data.currently.icon;
+  var detailsIcon = data.details.icon;
+  var todayDesc = data.details.today;
+  var tonightDesc = data.details.tonight;
+  var feelsLike = data.details.feels_like;
+  var humidity = data.details.humidity;
+  var visibility = data.details.visibility;
+  var visibility = data.details.visibility;
+  var uv = data.details.UV_Index;
+  var time1 = data.hourly[0].time;
```

Comment:
Starting here and below, there's a lot of repetition.  I'd try to build a for loop to get rid of most of this code.  Something like this:  ```for (var i = 0; i < 8; i++){
  id = i + 1;
  $(`#time-${id}`).html(data.hourly[i].time)
}```

Reply:
So now you're not declaring 8 variables and setting 8 different elements manually, but I would repeat this pattern for most of this code